infra/import_graph_test.py

```diff
@@ -0,0 +1,122 @@
+import unittest
+import json
+import os
+from collections import Counter
+
+import import_graph
+
+class TestTypeMatching(unittest.TestCase):
+    def test_node_type_matching(self):
+        # Testing the basic functionality. Cases don't need to be exhaustive;
+        # missing types will be caught during the whole-graph import tests.
+        example_issue = ["sourcecred", "github", "ISSUE", "sourcecred", "sourcecred", "34"]
+        example_repo = ["sourcecred", "github", "REPO", "sourcecred", "sourcecred"]
+        self.assertEqual(import_graph.node_type(example_issue), "github/issue")
+        self.assertEqual(import_graph.node_type(example_repo), "github/repo")
+        with (self.assertRaises(ValueError)):
+            import_graph.node_type(["non","existent", "node"])
+
+    def test_edge_type_matching(self):
+        # Testing the basic functionality. Cases don't need to be exhaustive;
+        # missing types will be caught during the whole-graph import tests.
+        example_has_parent= [
+          "sourcecred",
+          "github",
+          "HAS_PARENT",
+          "6",
+          "sourcecred",
+          "github",
+          "ISSUE",
+          "sourcecred",
+          "pm",
+          "1"
+        ]
+
+        example_authors= [
+          "sourcecred",
+          "github",
+          "AUTHORS",
+          "5",
+          "sourcecred",
+          "github",
+          "USERLIKE",
+          "USER",
+          "BrianLitwin",
+          "4",
+          "sourcecred",
+          "git",
+          "COMMIT",
+          "0cae9fa77c1d7d8b8fe3fe2d316a6782757862e4"
+        ]
+        self.assertEqual(import_graph.edge_type(example_has_parent), "github/hasParent")
+        self.assertEqual(import_graph.edge_type(example_authors), "github/authors")
+        with (self.assertRaises(ValueError)):
+            import_graph.edge_type(["non","existent", "edge"])
+
+def sample_graphs_directory():
+    wd = os.path.abspath(os.path.dirname(__file__))
```

Comment:
Ugh… is this really the best way to do this in Python? :-( With some
quick searching, I don’t see a better way, but it’s not exactly pretty
and it doesn’t scream robustness. No change requested, I guess.